### DISCUSSION TITLE
TypeLoadException after upgrade to 1.5

### DISCUSSION THREAD

**Discussion by DotCat1985** [OrchardCMS/OrchardCore#12763] [Q&A]
After upgrading **OrchardCore** to **1.5** in my solution and launching the CMS, I get **TypeLoadException** (see the details below).

An unhandled exception occurred while processing the request.

TypeLoadException: Could not load type 'OrchardCore.Autoroute.Services.AutorouteStateDocument' from assembly 'OrchardCore.Autoroute, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null'.

System.Signature.GetSignature(Void* pCorSig, int cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)

TypeLoadException: Could not load type 'OrchardCore.Autoroute.Services.AutorouteStateDocument' from assembly 'OrchardCore.Autoroute, Version=1.5.0.0, Culture=neutral, PublicKeyToken=null'.
System.Signature.GetSignature(Void* pCorSig, int cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
System.Reflection.RuntimeConstructorInfo.<get_Signature>g__LazyCreateSignature|19_0()
System.Reflection.RuntimeConstructorInfo.GetParameters()
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, int slot)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, bool throwIfCallSiteNotFound)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, int slot)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(Type serviceType)
System.Collections.Concurrent.ConcurrentDictionary<TKey, TValue>.GetOrAdd(TKey key, Func<TKey, TValue> valueFactory)
Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope.GetService(Type serviceType)
OrchardCore.Environment.Shell.Scope.ShellScopeServices.GetService(Type serviceType)
Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(IServiceProvider provider, Type serviceType)
Microsoft.AspNetCore.Mvc.Routing.DynamicControllerEndpointMatcherPolicy.ApplyAsync(HttpContext httpContext, CandidateSet candidates)
Microsoft.AspNetCore.Routing.Matching.DfaMatcher.SelectEndpointWithPoliciesAsync(HttpContext httpContext, IEndpointSelectorPolicy[] policies, CandidateSet candidateSet)
Microsoft.AspNetCore.Routing.EndpointRoutingMiddleware.<Invoke>g__AwaitMatch|8_1(EndpointRoutingMiddleware middleware, HttpContext httpContext, Task matchTask)
OrchardCore.Diagnostics.DiagnosticsStartupFilter+<>c__DisplayClass3_0+<<Configure>b__1>d.MoveNext()
Microsoft.AspNetCore.Diagnostics.StatusCodePagesMiddleware.Invoke(HttpContext context)
OrchardCore.ContentPreview.PreviewStartupFilter+<>c+<<Configure>b__1_1>d.MoveNext()
OrchardCore.Modules.ModularTenantRouterMiddleware.Invoke(HttpContext httpContext)
OrchardCore.Logging.SerilogTenantNameLoggingMiddleware.Invoke(HttpContext context)
OrchardCore.Modules.ModularTenantContainerMiddleware+<>c__DisplayClass4_0+<<Invoke>b__0>d.MoveNext()
OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func<ShellScope, Task> execute, bool activateShell)
OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func<ShellScope, Task> execute, bool activateShell)
OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func<ShellScope, Task> execute, bool activateShell)
OrchardCore.Environment.Shell.Scope.ShellScope.UsingAsync(Func<ShellScope, Task> execute, bool activateShell)
OrchardCore.Modules.ModularTenantContainerMiddleware.Invoke(HttpContext httpContext)
Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)

**Comment by Skrypt** [comment]
Nuget cache issue probably. The class is there in that assembly; it should work.

**Reply by DotCat1985** [reply]
The issue hasn't been solved after clearing Nuget cache. What could I do?

**Reply by jtkech** [reply]
This type was moved from `OrchardCore.Autoroute` to `OrchardCore.Autoroute.Core` with a different namespace. Looks like you still have a project / package that references the old type/signature. If it's a local project normally a full rebuild will output compilation errors, if it's from a tier package you may have some warnings, or you can check in your solution or on nuget its dependencies.

You may have to cleanup your solution, when working in sync with a github repo we can do `git clean -xdf` but that also delete locally all that is not in the repo, for example the `App_Data` folder, at least try to remove all /bin and /obj folders.

**Reply by Skrypt** [reply]
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore.Autoroute.Core/Model/AutorouteStateDocument.cs

**Reply by dsghi** [reply]
I had this error; it was due to another package; in my case, it was [Etch.OrchardCore.SEO](https://github.com/EtchUK/Etch.OrchardCore.SEO)

**Reply by peterkeating** [reply]
We need to upgrade the module to 1.5.0. Currently going through the process of upgrading our modules so shouldn't be too long 👍

**Reply by devlife** [reply]
I'm having this issue as well. Definitely looking forward to the update as I have an issue with OC 1.4 that I need to fix.

**Reply by peterkeating** [reply]
Module has been upgraded and available on NuGet 👍
